k8s/jenkins/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,250 @@
 actions:
-- name: dummy test
+- name: Jenkins service login
   bashTest:
     script: |-
-      exit 0
+      SERVICE="${APP_INSTANCE_NAME}-jenkins-ui"
+      IP=''
+      PORT=''
+      PASSWORD=''
+      FAILED=0
+
+      function wait_for_service_ip {
+        IP=$(kubectl get service "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.spec.clusterIP}")
+        if [[ $? -ne 0 ]]; then
+          IP=""
+        fi
+      }
+
+      function wait_for_service_port {
+        PORT=$(kubectl get service "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.spec.ports[0].port}")
+        if [[ $? -ne 0 ]]; then
+          PORT=""
+        fi
+      }
+
+      function get_jenkins_pod {
+        POD=$(kubectl get pod -oname \
+          --namespace "${NAMESPACE}" | \
+          sed -n "/\\/${APP_INSTANCE_NAME}-jenkins-deployment/s.pods\\?/..p")
+        if [[ $? -ne 0 ]]; then
+          POD=""
+        fi
+      }
+
+      function get_jenkins_admin_password {
+        PASSWORD=$(kubectl exec "${POD}" \
+          --namespace "${NAMESPACE}" \
+          cat /var/jenkins_home/secrets/initialAdminPassword)
+        if [[ $? -ne 0 ]]; then
+          PASSWORD=""
+        fi
+      }
+
+      function wait_for_ingress_healthy_state {
+        kubectl describe ingress "${SERVICE}" \
+          --namespace "${NAMESPACE}" | \
+          grep -q 'ingress.kubernetes.io/backends:.*{.*:"HEALTHY"}'
+      }
+
+      echo "Waiting for Service ip address."
+      wait_for_service_ip
+      while [[ "${IP}" == "" ]]; do
+        echo "Not yet -- Service IP"
+        sleep 20
+        wait_for_service_ip
+      done
+
+      echo "Waiting for Service port."
+      wait_for_service_port
+      while [[ "${PORT}" == "" ]]; do
+        echo "Not yet -- Service PORT"
+        sleep 20
+        wait_for_service_port
+      done
+
+      echo "Waiting for Jenkins Pod."
+      get_jenkins_pod
+      while [[ "${POD}" == "" ]]; do
+        echo "Not yet -- POD"
+        sleep 20
+        get_jenkins_pod
+      done
+
+      echo "Waiting for Jenkins Admin Password."
+      get_jenkins_admin_password
+      while [[ "${PASSWORD}" == "" ]]; do
+        echo "Not yet -- PASSWORD"
+        sleep 20
+        get_jenkins_admin_password
+      done
+
+      # test1 -- unathenticated curl on http://
```

Comment:
Please split below tests to separate actions. 

By design separate actions should test one thing. Also, testrunner has pretty nice expectations fields and output.

Spec: https://github.com/GoogleCloudPlatform/marketplace-testrunner/blob/master/specs/specs.go#L55

Reply:
These tests form a set of tests and should be in one place.